pkg/engine/internal/executor/translate_errors.go

```diff
@@ -0,0 +1,43 @@
+package executor
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/apache/arrow-go/v18/arrow"
+)
+
+func TranslateEOF(pipeline Pipeline) Pipeline {
+	return translateEOFPipeline{pipeline}
+}
+
+type translateEOFPipeline struct {
```

Comment:
Does this need to be a pipeline? Seems like it could be a standard function wrapping an error